config/main.py

```diff
@@ -1797,8 +1798,15 @@ def load_minigraph(db, no_service_restart, traffic_shift_away):
             click.secho("[WARNING] Golden configuration may override Traffic-shift-away state. Please execute TSC to check the current System mode")
 
     # Load golden_config_db.json
-    if os.path.isfile(DEFAULT_GOLDEN_CONFIG_DB_FILE):
-        override_config_by(DEFAULT_GOLDEN_CONFIG_DB_FILE)
+    if golden_config_path:
+        if not os.path.isfile(golden_config_path):
+            click.secho("Cannot find '{}'!".format(golden_config_path),
+                        fg='magenta')
+            raise click.Abort()
+        override_config_by(golden_config_path)
+    else:
```

Comment:
>else [](http://example.com/codeflow?start=4&length=4)

You are treating below 2 usages the same.
1. `config load_minigraph --golden_config_path`
2. `config load_minigraph`

The user expectation may be
1. User want to override, but do not provide a gulden config path, so just

Reply:
I am forcing users to add an argument when add `--golden_config_path`.
`config load_minigraph --golden_config_path` will return an error: 
`Error: --golden_config_path option requires an argument`

> 2. User does not want to override.

Are we chaning the code behavior? Previously, `load_minigraph` will check DEFAULT_GOLDEN_PATH and override if it exists. Do you mean to change to override only when `--golden_config_path` is set?

Reply:
Let's merge current PR, backport, ask client to explicitly use the option, and you submit another PR to achieve the expected behavior.